include/container.hpp

```diff
@@ -95,20 +93,22 @@ class container
   container(int *buf) : buffer(buf){};
 
   container(const container &other)
-      : container(copy(other.buffer, other.capacity)){};
+      : current_size(other.current_size),
+        buffer(copy(other.buffer, other.capacity)) {}
 
   container(container &&other) noexcept
       : buffer(std::exchange(other.buffer, nullptr)){};
 
   container &operator=(const container &other) {
-    if (this == &other)
-      return *this;
-
-    container temp(other);
-    std::swap(buffer, temp.buffer);
-
+    if (this != &other) {
+      if (current_size != other.current_size) {
+        buffer = allocator.allocate(other.capacity);
+        current_size = other.current_size;
+      }
+      std::copy(&other.buffer[0], &other.buffer[0] + current_size, &buffer[0]);
```

Comment:
this is ok for simple types but will not work on non trivial with custom copy logic.
Would expect copy(other.buffer, other.size, other.capacity)